### DISCUSSION TITLE
How to use `IPyObjectDecoder` to convert a python list to a .NET List<string>

### DISCUSSION THREAD

**Discussion by awa5114** [pythonnet/pythonnet#2277] [Q&A]
Could someone provide an example on how to do this? I want to pass a python list to a CLR class imported in my Python script and let .NET handle all the conversion related work. The priority here is really to keep my python script clean and do all the heavy lifting from .NET.  Here's what I've tried so far:

```cs
public class PythonListDecoder : IPyObjectDecoder
{
    bool CanDecode(PyObject objectType, Type targetType)
    {
        return targetType == typeof(List<string>);
    }

    bool TryDecode<T>(PyObject obj, out T value)
    {
    }
}
```

But I don't really know what needs to be done from there and haven't been able to find sufficient documentation to proceed.

**Comment by JakeJP** [comment]
Did you already try ListDecoder? It enables IList&lt;string&gt; convertible with python list as far as I know.
If you have to stick to List&lt;string&gt;, good example code is in ListDecoder.cs.

Python.Runtime.Codecs.ListDecoder.Register()
Python.Runtime.Codecs.SequenceDecoder.Register()
Python.Runtime.Codecs.IterableDecoder.Register()
